tests/conftest.py

```diff
@@ -303,26 +303,41 @@ def _slurm_not_installed():
 def _subprocess_setup(request, cfg=None, uwsgi=False):
     """setup the supervisor"""
     import os
+    import sirepo.job
+    import sirepo.pkcli.service
     from pykern.pkcollections import PKDict
 
+    ports = PKDict()
+
+    # different port than default so can run tests when supervisor running
+    def _next_conftest_port(port_key, port_pkg):
+        p = ports.get(port_key)
+        if p:
+            ports[port_key] = p + 1
+        else:
+            ports[port_key] = getattr(port_pkg, port_key) + 100
+        return str(ports[port_key])
+
     sbatch_module = "sbatch" in request.module.__name__
     env = PKDict(os.environ)
     if not cfg:
         cfg = PKDict()
     from pykern import pkunit
     from pykern import pkio
 
-    # different port than default so can run tests when supervisor running
-    p = "8101"
+    p = _next_conftest_port("DEFAULT_PORT", sirepo.job)
     cfg.pkupdate(
         PYKERN_PKDEBUG_WANT_PID_TIME="1",
         SIREPO_PKCLI_JOB_SUPERVISOR_IP=_LOCALHOST,
         SIREPO_PKCLI_JOB_SUPERVISOR_PORT=p,
         SIREPO_SRDB_ROOT=str(pkio.mkdir_parent(pkunit.work_dir().join("db"))),
     )
     if uwsgi:
-        cfg.SIREPO_PKCLI_SERVICE_PORT = "8102"
-        cfg.SIREPO_PKCLI_SERVICE_NGINX_PROXY_PORT = "8180"
+        cfg.SIREPO_PKCLI_SERVICE_PORT = _next_conftest_port("DEFAULT_PORT", sirepo.job)
```

Comment:
It could be argued that this name is incorrect. It should be SIREPO_PKCLI_SERVICE_HTTP_PORT and separately *_UWSGI_PORT.  Having explicit values would simplify things. The PORT_DEFAULTS would be the same, which would argue against this...